src/components/UserInfo/UserInfo.tsx

```diff
@@ -1 +1,20 @@
-export const UserInfo = () => {};
+import React from 'react';
+
+type UserInfoProps = {
+  user: {
+    name: string;
+    email?: string;
+  };
+};
+
+export const UserInfo: React.FC<UserInfoProps> = ({ user }) => {
+  if (!user || !user.email) {
+    return null;
+  }
```

Comment:
```suggestion

```
looks redundant to me
<img width="405" alt="image" src="https://github.com/user-attachments/assets/d8e03f53-681c-4cf0-9b9a-9434cc79aea1">
you render this component only when the user exists